tfchain_bridge/pkg/stellar/stellar.go

```diff
@@ -327,79 +313,51 @@ func (w *StellarWallet) MonitorBridgeAccountAndMint(ctx context.Context, mintFn
 			if effect.GetAccount() != w.config.StellarBridgeAccount {
 				continue
 			}
-			if effect.GetType() == "account_credited" {
-				creditedEffect := effect.(horizoneffects.AccountCredited)
-				if creditedEffect.Asset.Code != asset[0] && creditedEffect.Asset.Issuer != asset[1] {
+
+			if effect.GetType() != "account_credited" {
+				continue
+			}
+
+			creditedEffect := effect.(horizoneffects.AccountCredited)
+			if creditedEffect.Asset.Code != asset[0] && creditedEffect.Asset.Issuer != asset[1] {
+				continue
+			}
+			parsedAmount, err := amount.ParseInt64(creditedEffect.Amount)
+			if err != nil {
+				continue
+			}
+			depositedAmount := big.NewInt(int64(parsedAmount))
+
+			ops, err := w.getOperationEffect(tx.Hash)
+			if err != nil {
+				continue
+			}
+
+			for _, op := range ops.Embedded.Records {
+				if op.GetType() != "payment" {
 					continue
 				}
-				parsedAmount, err := amount.ParseInt64(creditedEffect.Amount)
-				if err != nil {
+
+				paymentOpation := op.(operations.Payment)
+				if paymentOpation.To != w.config.StellarBridgeAccount {
 					continue
 				}
 
-				depositedAmount := big.NewInt(int64(parsedAmount))
-
-				err = mintFn(tx.Account, depositedAmount, tx.Hash)
+				err := mintFn(paymentOpation.From, depositedAmount, tx)
```

Comment:
I think this should not be depositedAmount, but rather the amount of this payment operation (`depostedAmount.Amount`). Right now, if a tx includes 2 payments (say 100 TFT each), depositedamount will be 200. Since this loop goes through every payment and mints the depositedamount, it would effectively mint 2 times 200 TFT instead of 2 times 100 TFT.